### PR TITLE
fix: adds support for tap 19

### PR DESCRIPTION
**Pull request by reggi**
* currently `tap.nyc-args` is only removed for 18 not 19 😢 
* `tap-snapshots` is ignored by default
* adds `tap.show-full-coverage` for > 17
* updates snaps to use new `>` delimiter syntax
* `t.mock` -> `t.mockRequire`
* had to reconstruct fixtureName to match old one because fixtures have different names now, ideally, would like to rm all this code and use https://github.com/npm/tap-nock and rename nock snapshots

### CODE REVIEW

File: lib/content/package-json.hbs [hunk 1]
```diff
@@ -40,10 +40,20 @@
   },
   "templateVersion": {{{ del }}},
   "standard": {{{ del }}},
+  "nyc": {{#if tapOver17}}{
+    "exclude": {{{ json workspaceGlobs }}}
```
Comment (wraithgar) [inline lib/content/package-json.hbs:44]: Is this related to some new change in tap 19?

Comment (reggi) [reply lib/content/package-json.hbs:44]: yes

**Review by wraithgar — APPROVED** [review]

**Comment by wraithgar** [comment]
Oh hrm, didn't see that tests were red still.  Of course that needs to work first.

**Comment by reggi** [comment]
for some reason `release-please.js` got less coverage 🤔 

```
lib/release               |   78.41 |    96.59 |   88.37 |   78.41 |                                      
  changelog.js             |     100 |      100 |     100 |     100 |                                      
  node-workspace-format.js |    9.27 |    66.66 |   33.33 |    9.27 | 10-97                                
  release-manager.js       |   99.26 |    97.95 |     100 |   99.26 | 176-177                              
  release-please.js        |   48.82 |     62.5 |   57.14 |   48.82 | 50,65-73,84-85,89-141,144-148,151-167
  util.js                  |     100 |      100 |     100 |     100 |                      
  ```

**Comment by reggi** [comment]
Blocked by https://github.com/tapjs/tapjs/issues/1027

**Comment by reggi** [comment]
Blocked by https://github.com/tapjs/tapjs/issues/950 because tap@19 doesn't work on 20.5.0 😢 not sure what to do here, do we drop support for this node version?

**Comment by reggi** [comment]
I've made pr's which could potentially make tap truly support what it says `>=16 `

https://github.com/isaacs/resolve-import/pull/5
https://github.com/tapjs/tapjs/pull/1028

But as it currently stands tap truly is 
```
>=16.17.0 <=16.20.2||>=18.6.0 <=18.20.3||>=19.0.0 <=19.5.0||>=20.6.0
// or
>=16.17.0 <=16.20.2||>=18.6.0 <=19.5.0||>=20.6.0
```
and we still use node 20.5 to test 😢

**Comment by reggi** [comment]
I've added a small demo here of tap@19 working on node 16-22  here it patch-package 😞 

https://github.com/reggi/patch-tap/tree/main/patches
